### DISCUSSION TITLE
Installation issues

### DISCUSSION THREAD

**Discussion by Slejpnir** [GrandMoff100/HomeAssistantAPI#144] [Q&A]
Hello, I am trying to install the package on my Raspberry Pi 4 with RaspberryOS installed.
But I am receiving the error

```
ERROR: Command errored out with exit status 1:
   command: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-hb6c01eo/cchardet_e0832ca940d34667ac4ac7c6724a8cd0/setup.py'"'"'; __file__='"'"'/tmp/pip-install-hb6c01eo/cchardet_e0832ca940d34667ac4ac7c6724a8cd0/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-33fg3mwy
       cwd: /tmp/pip-install-hb6c01eo/cchardet_e0832ca940d34667ac4ac7c6724a8cd0/
  Complete output (24 lines):
  cythonize: ['src/cchardet/_cchardet.pyx']
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-aarch64-3.9
  creating build/lib.linux-aarch64-3.9/cchardet
  copying src/cchardet/version.py -> build/lib.linux-aarch64-3.9/cchardet
  copying src/cchardet/__init__.py -> build/lib.linux-aarch64-3.9/cchardet
  running build_ext
  building 'cchardet._cchardet' extension
  creating build/temp.linux-aarch64-3.9
  creating build/temp.linux-aarch64-3.9/src
  creating build/temp.linux-aarch64-3.9/src/cchardet
  creating build/temp.linux-aarch64-3.9/src/ext
  creating build/temp.linux-aarch64-3.9/src/ext/uchardet
  creating build/temp.linux-aarch64-3.9/src/ext/uchardet/src
  creating build/temp.linux-aarch64-3.9/src/ext/uchardet/src/LangModels
  aarch64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -ffile-prefix-map=/build/python3.9-PN012d/python3.9-3.9.2=. -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -ffile-prefix-map=/build/python3.9-PN012d/python3.9-3.9.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -Isrc/ext/uchardet/src -I/usr/include/python3.9 -c src/cchardet/_cchardet.cpp -o build/temp.linux-aarch64-3.9/src/cchardet/_cchardet.o
  aarch64-linux-gnu-gcc: internal compiler error: Segmentation fault signal terminated program cc1plus
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-10/README.Bugs> for instructions.
  error: command '/usr/bin/aarch64-linux-gnu-gcc' failed with exit code 4
  ----------------------------------------
  ERROR: Failed building wheel for cchardet
  Running setup.py clean for cchardet
Failed to build cchardet
Installing collected packages: cchardet
    Running setup.py install for cchardet ... error
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-hb6c01eo/cchardet_e0832ca940d34667ac4ac7c6724a8cd0/setup.py'"'"'; __file__='"'"'/tmp/pip-install-hb6c01eo/cchardet_e0832ca940d34667ac4ac7c6724a8cd0/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-mvk7n50n/install-record.txt --single-version-externally-managed --user --prefix= --compile --install-headers /home/slejpnir/.local/include/python3.9/cchardet
         cwd: /tmp/pip-install-hb6c01eo/cchardet_e0832ca940d34667ac4ac7c6724a8cd0/
    Complete output (24 lines):
    cythonize: ['src/cchardet/_cchardet.pyx']
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-aarch64-3.9
    creating build/lib.linux-aarch64-3.9/cchardet
    copying src/cchardet/version.py -> build/lib.linux-aarch64-3.9/cchardet
    copying src/cchardet/__init__.py -> build/lib.linux-aarch64-3.9/cchardet
    running build_ext
    building 'cchardet._cchardet' extension
    creating build/temp.linux-aarch64-3.9
    creating build/temp.linux-aarch64-3.9/src
    creating build/temp.linux-aarch64-3.9/src/cchardet
    creating build/temp.linux-aarch64-3.9/src/ext
    creating build/temp.linux-aarch64-3.9/src/ext/uchardet
    creating build/temp.linux-aarch64-3.9/src/ext/uchardet/src
    creating build/temp.linux-aarch64-3.9/src/ext/uchardet/src/LangModels
    aarch64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -ffile-prefix-map=/build/python3.9-PN012d/python3.9-3.9.2=. -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -ffile-prefix-map=/build/python3.9-PN012d/python3.9-3.9.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -Isrc/ext/uchardet/src -I/usr/include/python3.9 -c src/cchardet/_cchardet.cpp -o build/temp.linux-aarch64-3.9/src/cchardet/_cchardet.o
    aarch64-linux-gnu-gcc: internal compiler error: Segmentation fault signal terminated program cc1plus
    Please submit a full bug report,
    with preprocessed source if appropriate.
    See <file:///usr/share/doc/gcc-10/README.Bugs> for instructions.
    error: command '/usr/bin/aarch64-linux-gnu-gcc' failed with exit code 4
    ----------------------------------------
ERROR: Command errored out with exit status 1: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-hb6c01eo/cchardet_e0832ca940d34667ac4ac7c6724a8cd0/setup.py'"'"'; __file__='"'"'/tmp/pip-install-hb6c01eo/cchardet_e0832ca940d34667ac4ac7c6724a8cd0/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-mvk7n50n/install-record.txt --single-version-externally-managed --user --prefix= --compile --install-headers /home/slejpnir/.local/include/python3.9/cchardet Check the logs for full command output.

```
I will be very thankful for any assistance.

**Comment by GrandMoff100** [comment]
The installation is failing because it's trying to compile `cchardet`, an optional aiohttp speedup library, which is written for a different compiler on another platform. This might actually be an issue but I need to do more research on aiohttp dependencies to figure that out.


As a short term solution, @Slejpnir, try installing aiohttp by itself with `pip install aiohttp` and report back whether or not that fails. If it doesn't fail then try `pip install homeassistant_api` again and report back if that fails or not.

**Reply by Slejpnir** [reply]
pip install aiohttp didn't fail, but pip install homeassistant_api did fail anyway afterwards.

**Reply by GrandMoff100** [reply]
Could you send the logs of both commands?

**Reply by Slejpnir** [reply]
pip install aiohttp
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Requirement already satisfied: aiohttp in ./.local/lib/python3.9/site-packages (3.8.3)
Requirement already satisfied: multidict<7.0,>=4.5 in ./.local/lib/python3.9/site-packages (from aiohttp) (6.0.4)
Requirement already satisfied: charset-normalizer<3.0,>=2.0 in ./.local/lib/python3.9/site-packages (from aiohttp) (2.1.1)
Requirement already satisfied: attrs>=17.3.0 in ./.local/lib/python3.9/site-packages (from aiohttp) (21.4.0)
Requirement already satisfied: yarl<2.0,>=1.0 in ./.local/lib/python3.9/site-packages (from aiohttp) (1.8.2)
Requirement already satisfied: aiosignal>=1.1.2 in ./.local/lib/python3.9/site-packages (from aiohttp) (1.3.1)
Requirement already satisfied: frozenlist>=1.1.1 in ./.local/lib/python3.9/site-packages (from aiohttp) (1.3.3)
Requirement already satisfied: async-timeout<5.0,>=4.0.0a3 in ./.local/lib/python3.9/site-packages (from aiohttp) (4.0.2)
Requirement already satisfied: idna>=2.0 in ./.local/lib/python3.9/site-packages (from yarl<2.0,>=1.0->aiohttp) (3.4)

**Reply by Slejpnir** [reply]
pip install homeassistant_api
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Collecting homeassistant_api
  Using cached https://www.piwheels.org/simple/homeassistant-api/homeassistant_api-4.0.1-py3-none-any.whl (33 kB)
Requirement already satisfied: aiohttp<4.0.0,>=3.8.1 in ./.local/lib/python3.9/site-packages (from homeassistant_api) (3.8.3)
Collecting requests-cache<0.10.0,>=0.9.2
  Using cached https://www.piwheels.org/simple/requests-cache/requests_cache-0.9.8-py3-none-any.whl (48 kB)
Collecting aiohttp-client-cache<0.7.0,>=0.6.1
  Using cached https://www.piwheels.org/simple/aiohttp-client-cache/aiohttp_client_cache-0.6.1-py3-none-any.whl (31 kB)
Requirement already satisfied: requests<3.0.0,>=2.27.1 in ./.local/lib/python3.9/site-packages (from homeassistant_api) (2.28.1)
Collecting simplejson<4.0.0,>=3.17.6
  Using cached simplejson-3.18.1-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (137 kB)
Collecting pydantic<=1.9.0
  Using cached pydantic-1.9.0-py3-none-any.whl (140 kB)
Requirement already satisfied: aiosignal>=1.1.2 in ./.local/lib/python3.9/site-packages (from aiohttp<4.0.0,>=3.8.1->homeassistant_api) (1.3.1)
Requirement already satisfied: yarl<2.0,>=1.0 in ./.local/lib/python3.9/site-packages (from aiohttp<4.0.0,>=3.8.1->homeassistant_api) (1.8.2)
Requirement already satisfied: frozenlist>=1.1.1 in ./.local/lib/python3.9/site-packages (from aiohttp<4.0.0,>=3.8.1->homeassistant_api) (1.3.3)
Requirement already satisfied: multidict<7.0,>=4.5 in ./.local/lib/python3.9/site-packages (from aiohttp<4.0.0,>=3.8.1->homeassistant_api) (6.0.4)
Requirement already satisfied: attrs>=17.3.0 in ./.local/lib/python3.9/site-packages (from aiohttp<4.0.0,>=3.8.1->homeassistant_api) (21.4.0)
Requirement already satisfied: charset-normalizer<3.0,>=2.0 in ./.local/lib/python3.9/site-packages (from aiohttp<4.0.0,>=3.8.1->homeassistant_api) (2.1.1)
Requirement already satisfied: async-timeout<5.0,>=4.0.0a3 in ./.local/lib/python3.9/site-packages (from aiohttp<4.0.0,>=3.8.1->homeassistant_api) (4.0.2)
Collecting itsdangerous
  Using cached https://www.piwheels.org/simple/itsdangerous/itsdangerous-2.1.2-py3-none-any.whl (15 kB)
Collecting python-forge<19.0,>=18.6
  Using cached python_forge-18.6.0-py35-none-any.whl (31 kB)
Collecting url-normalize<2.0,>=1.4
  Using cached https://www.piwheels.org/simple/url-normalize/url_normalize-1.4.3-py2.py3-none-any.whl (6.8 kB)
Collecting aiodns
  Using cached https://www.piwheels.org/simple/aiodns/aiodns-3.0.0-py3-none-any.whl (5.0 kB)
Collecting cchardet
  Using cached cchardet-2.1.7.tar.gz (653 kB)
Collecting Brotli
  Using cached Brotli-1.0.9-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (2.6 MB)
Collecting typing-extensions>=3.7.4.3
  Using cached https://www.piwheels.org/simple/typing-extensions/typing_extensions-4.4.0-py3-none-any.whl (26 kB)
Requirement already satisfied: idna<4,>=2.5 in ./.local/lib/python3.9/site-packages (from requests<3.0.0,>=2.27.1->homeassistant_api) (3.4)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in ./.local/lib/python3.9/site-packages (from requests<3.0.0,>=2.27.1->homeassistant_api) (1.26.14)
Requirement already satisfied: certifi>=2017.4.17 in ./.local/lib/python3.9/site-packages (from requests<3.0.0,>=2.27.1->homeassistant_api) (2022.12.7)
Collecting cattrs>=22.2
  Using cached https://www.piwheels.org/simple/cattrs/cattrs-22.2.0-py3-none-any.whl (35 kB)
Collecting appdirs>=1.4.4
  Using cached https://www.piwheels.org/simple/appdirs/appdirs-1.4.4-py2.py3-none-any.whl (9.6 kB)
Requirement already satisfied: exceptiongroup in ./.local/lib/python3.9/site-packages (from cattrs>=22.2->requests-cache<0.10.0,>=0.9.2->homeassistant_api) (1.1.0)
Requirement already satisfied: six in ./.local/lib/python3.9/site-packages (from url-normalize<2.0,>=1.4->aiohttp-client-cache<0.7.0,>=0.6.1->homeassistant_api) (1.16.0)
Requirement already satisfied: pycares>=4.0.0 in ./.local/lib/python3.9/site-packages (from aiodns->aiohttp<4.0.0,>=3.8.1->homeassistant_api) (4.3.0)
Requirement already satisfied: cffi>=1.5.0 in ./.local/lib/python3.9/site-packages (from pycares>=4.0.0->aiodns->aiohttp<4.0.0,>=3.8.1->homeassistant_api) (1.15.0)
Requirement already satisfied: pycparser in ./.local/lib/python3.9/site-packages (from cffi>=1.5.0->pycares>=4.0.0->aiodns->aiohttp<4.0.0,>=3.8.1->homeassistant_api) (2.21)
Building wheels for collected packages: cchardet
  Building wheel for cchardet (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-p3no2_02/cchardet_5405843242434c32af2a5acc99e84e46/setup.py'"'"'; __file__='"'"'/tmp/pip-install-p3no2_02/cchardet_5405843242434c32af2a5acc99e84e46/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-a3depkcn
       cwd: /tmp/pip-install-p3no2_02/cchardet_5405843242434c32af2a5acc99e84e46/
  Complete output (24 lines):
  cythonize: ['src/cchardet/_cchardet.pyx']
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-aarch64-3.9
  creating build/lib.linux-aarch64-3.9/cchardet
  copying src/cchardet/version.py -> build/lib.linux-aarch64-3.9/cchardet
  copying src/cchardet/__init__.py -> build/lib.linux-aarch64-3.9/cchardet
  running build_ext
  building 'cchardet._cchardet' extension
  creating build/temp.linux-aarch64-3.9
  creating build/temp.linux-aarch64-3.9/src
  creating build/temp.linux-aarch64-3.9/src/cchardet
  creating build/temp.linux-aarch64-3.9/src/ext
  creating build/temp.linux-aarch64-3.9/src/ext/uchardet
  creating build/temp.linux-aarch64-3.9/src/ext/uchardet/src
  creating build/temp.linux-aarch64-3.9/src/ext/uchardet/src/LangModels
  aarch64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -ffile-prefix-map=/build/python3.9-PN012d/python3.9-3.9.2=. -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -ffile-prefix-map=/build/python3.9-PN012d/python3.9-3.9.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -Isrc/ext/uchardet/src -I/usr/include/python3.9 -c src/cchardet/_cchardet.cpp -o build/temp.linux-aarch64-3.9/src/cchardet/_cchardet.o
  aarch64-linux-gnu-gcc: internal compiler error: Segmentation fault signal terminated program cc1plus
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-10/README.Bugs> for instructions.
  error: command '/usr/bin/aarch64-linux-gnu-gcc' failed with exit code 4
  ----------------------------------------
  ERROR: Failed building wheel for cchardet
  Running setup.py clean for cchardet
Failed to build cchardet
Installing collected packages: cchardet, Brotli, aiodns, url-normalize, typing-extensions, python-forge, itsdangerous, cattrs, appdirs, simplejson, requests-cache, pydantic, aiohttp-client-cache, homeassistant-api
    Running setup.py install for cchardet ... error
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-p3no2_02/cchardet_5405843242434c32af2a5acc99e84e46/setup.py'"'"'; __file__='"'"'/tmp/pip-install-p3no2_02/cchardet_5405843242434c32af2a5acc99e84e46/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-tznph50t/install-record.txt --single-version-externally-managed --user --prefix= --compile --install-headers /home/slejpnir/.local/include/python3.9/cchardet
         cwd: /tmp/pip-install-p3no2_02/cchardet_5405843242434c32af2a5acc99e84e46/
    Complete output (24 lines):
    cythonize: ['src/cchardet/_cchardet.pyx']
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-aarch64-3.9
    creating build/lib.linux-aarch64-3.9/cchardet
    copying src/cchardet/version.py -> build/lib.linux-aarch64-3.9/cchardet
    copying src/cchardet/__init__.py -> build/lib.linux-aarch64-3.9/cchardet
    running build_ext
    building 'cchardet._cchardet' extension
    creating build/temp.linux-aarch64-3.9
    creating build/temp.linux-aarch64-3.9/src
    creating build/temp.linux-aarch64-3.9/src/cchardet
    creating build/temp.linux-aarch64-3.9/src/ext
    creating build/temp.linux-aarch64-3.9/src/ext/uchardet
    creating build/temp.linux-aarch64-3.9/src/ext/uchardet/src
    creating build/temp.linux-aarch64-3.9/src/ext/uchardet/src/LangModels
    aarch64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -ffile-prefix-map=/build/python3.9-PN012d/python3.9-3.9.2=. -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -ffile-prefix-map=/build/python3.9-PN012d/python3.9-3.9.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -Isrc/ext/uchardet/src -I/usr/include/python3.9 -c src/cchardet/_cchardet.cpp -o build/temp.linux-aarch64-3.9/src/cchardet/_cchardet.o
    aarch64-linux-gnu-gcc: internal compiler error: Segmentation fault signal terminated program cc1plus
    Please submit a full bug report,
    with preprocessed source if appropriate.
    See <file:///usr/share/doc/gcc-10/README.Bugs> for instructions.
    error: command '/usr/bin/aarch64-linux-gnu-gcc' failed with exit code 4
    ----------------------------------------
ERROR: Command errored out with exit status 1: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-p3no2_02/cchardet_5405843242434c32af2a5acc99e84e46/setup.py'"'"'; __file__='"'"'/tmp/pip-install-p3no2_02/cchardet_5405843242434c32af2a5acc99e84e46/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-tznph50t/install-record.txt --single-version-externally-managed --user --prefix= --compile --install-headers /home/slejpnir/.local/include/pytho
